cmd/sink/filters/filters.go

```diff
@@ -332,8 +332,26 @@ func (f *Filters) handleFsEvent(logger *log.Logger, event fsnotify.Event) {
 		logger.Println("Removing filters for namespace:", fileName)
 		f.deleteNamespaceFilters(fileName)
 		logger.Println("Removed filters for namespace:", fileName)
+	case fsnotify.Chmod:
+		// fsnotify.Chmod needs to reload all of the filters. If we don't do this, we won't be able to detect the
+		// updates for KubeArchiveConfigs that are updated because the files that we are watching are symlinks
+		logger.Println("received fsnotify.Chmod event. Updating all filters")
+		filePaths, err := files.DirectoryFiles(f.path)
+		if err != nil {
+			logger.Printf("Could not get all files in directory %s: %s\n", f.path, err)
+			break
+		}
+		globalArchive, globalDelete, globalArchiveOnDelete, err := getGlobalCelExprs(filePaths)
+		if err != nil && !errors.Is(err, ErrNoGlobal) {
+			logger.Println("Could not read global filters:", err)
+			break
+		}
+		err = f.changeGlobalFilters(filePaths, globalArchive, globalDelete, globalArchiveOnDelete)
+		if err != nil {
+			logger.Println("Problem updating filters:", err)
+		}
+		logger.Println("Successfully updated filters")
```

Comment:
I haven't been able to figure it out exactly, but it seems like there's a lot of duplicated code in the `switch` statement.  Is there a way to simplify this?  It seems like `chmod` is similar to `write`.

This section from https://github.com/fsnotify/fsnotify concerns me:
```
Why do I get many Chmod events?

Some programs may generate a lot of attribute changes; for example
Spotlight on macOS, anti-virus programs, backup applications, and
some others are known to do this. As a rule, it's typically best to
ignore Chmod events. They're often not useful, and tend to cause problems.
```
Why do we have to use `chmod` here? Is it because these are all symlinks?

Reply:
I'm no longer using `fsnotify`, because it does not work with the two layers of symlinks that kubernetes creates when mounting a `ConfigMap`. I think the new logic is much simpler